### PR TITLE
remove dependency of flash_attn when use_flash_attn is set to false

### PR DESCRIPTION
**Pull request by sallyjunjun**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

v100 is not supported in flash attention. In order to support v100 users using InternEvo, we remove dependency of flash attention module when use_flash_attn is set to false.

## Modification

all flash attention related code.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: internlm/model/linear.py [hunk 1]
```diff
@@ -114,7 +118,47 @@ def forward(self, input):  # pylint: disable=W0622
         )
 
 
-class ColumnParallelLinearTorch(ColumnParallelLinear):
+class ColumnParallelLinearTorch(nn.Linear):
```
Comment (huangting4201) [inline internlm/model/linear.py:153]: mark一下，如果这里变动的话，isp算法set_parallel_attr_for_param_groups函数需要相应更新

Comment (sallyjunjun) [reply internlm/model/linear.py:153]: 这里只是将继承类更换了，将init函数实现了一下，主题逻辑没有变动。

File: internlm/model/embedding.py [hunk 2]
```diff
@@ -63,6 +60,22 @@ def forward(self, input_: Tensor) -> Tensor:
         return output
 
 
+def apply_rotary_torch(x1, x2, cos, sin, conj):
```
Comment (sunpengsdu) [inline internlm/model/embedding.py]: 这段代码是从哪适配的？

Comment (sallyjunjun) [reply internlm/model/embedding.py]: 根据flash attention里面rotary_emb.apply_rotary 函数的处理逻辑，实现了不使用cuda计算的版本。

File: internlm/model/embedding.py [hunk 3]
```diff
@@ -63,11 +60,280 @@ def forward(self, input_: Tensor) -> Tensor:
         return output
```
Comment (sunpengsdu) [inline internlm/model/embedding.py:62]: TODO: add ref link where the implementation from

**Comment by sunpengsdu** [comment]
should have some test cases?

**Review by sunpengsdu — APPROVED** [review]

**Pull request merged**